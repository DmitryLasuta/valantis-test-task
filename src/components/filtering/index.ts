export * from './Filtering';
export * from './FilterField';
