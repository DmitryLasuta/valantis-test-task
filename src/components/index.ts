export * from './pagination';
export * from './productCard';
