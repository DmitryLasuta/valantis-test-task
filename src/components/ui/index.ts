export * from './grid';
export * from './pageLayout';
export * from './theFooter';
