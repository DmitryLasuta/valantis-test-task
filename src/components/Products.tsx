import { Grid } from './ui';
import { ProductCard } from './ui/productCard';
import { QUERY_KEYS } from '@/shared/constants';
import { storeAPI } from '@/shared/api/store';
import { useQuery } from '@tanstack/react-query';

const Products = ({ identifiers }: { identifiers: string[] }) => {
  const {
    data: products,
    isPending,
    isError,
  } = useQuery({
    queryKey: [QUERY_KEYS.PRODUCTS, identifiers],
    queryFn: () => storeAPI.getProducts(identifiers),
    staleTime: 1000 * 60 * 5,
  });

  if (!products || isPending) return <div>Loading...</div>;
  if (isError) return <div>Error</div>;

  return (
    <Grid>
      {products.map(product => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { Products };
