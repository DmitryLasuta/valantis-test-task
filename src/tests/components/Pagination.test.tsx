import { describe, expect, test, vi } from 'vitest';
import { fireEvent, render } from '@testing-library/react';

import { Pagination } from '@/components';

describe('Pagination', () => {
  test('should render correctly with initial props', () => {
    const onPageChange = vi.fn();
    const { getByLabelText, baseElement } = render(
      <Pagination currentPage={1} totalPages={5} onPageChange={onPageChange} />
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
    expect(getByLabelText('Go to previous page')).toBeDisabled();
    expect(getByLabelText('Go to next page')).not.toBeDisabled();
  });

  test('should call onPageChange with the correct page when next button is clicked', () => {
    const onPageChange = vi.fn();
    const { getByLabelText } = render(<Pagination currentPage={1} totalPages={5} onPageChange={onPageChange} />);

    fireEvent.click(getByLabelText('Go to next page'));
    expect(onPageChange).toHaveBeenCalledWith(2);
  });

  test('should call onPageChange with the correct page when a page number is clicked', () => {
    const onPageChange = vi.fn();
    const { getByLabelText } = render(<Pagination currentPage={1} totalPages={5} onPageChange={onPageChange} />);

    fireEvent.click(getByLabelText('Go to page 3'));
    expect(onPageChange).toHaveBeenCalledWith(3);
  });

  test('should disable next button when currentPage is equal to totalPages', () => {
    const onPageChange = vi.fn();
    const { getByLabelText } = render(<Pagination currentPage={5} totalPages={5} onPageChange={onPageChange} />);

    expect(getByLabelText('Go to next page')).toBeDisabled();
  });

  test('should disable previous button when currentPage is equal to 1', () => {
    const onPageChange = vi.fn();
    const { getByLabelText } = render(<Pagination currentPage={1} totalPages={5} onPageChange={onPageChange} />);

    expect(getByLabelText('Go to previous page')).toBeDisabled();
  });

  test('should call onPageChange with the correct page when previous button is clicked', () => {
    const onPageChange = vi.fn();
    const { getByLabelText } = render(<Pagination currentPage={2} totalPages={5} onPageChange={onPageChange} />);

    fireEvent.click(getByLabelText('Go to previous page'));
    expect(onPageChange).toHaveBeenCalledWith(1);
  });
});
