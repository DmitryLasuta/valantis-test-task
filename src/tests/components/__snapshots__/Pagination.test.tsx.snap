// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should render correctly with initial props 1`] = `
<body>
  <div>
    <nav
      aria-label="Pagination"
      class="_pagination_b64421"
    >
      <button
        aria-label="Go to previous page"
        class="_pagination__button_b64421"
        disabled=""
        type="button"
      >
        &lt;
      </button>
      <ul
        class="_pagination__pages-list_b64421"
      >
        <li>
          <button
            aria-label="Go to page 1"
            class="_pagination__button_b64421"
            disabled=""
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="_pagination__button_b64421"
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="_pagination__button_b64421"
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="_pagination__button_b64421"
            type="button"
          >
            4
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="_pagination__button_b64421"
            type="button"
          >
            5
          </button>
        </li>
      </ul>
      <button
        aria-label="Go to next page"
        class="_pagination__button_b64421"
        type="button"
      >
        &gt;
      </button>
      <p>
        Page 
        1
         of 
        5
      </p>
    </nav>
  </div>
</body>
`;
